src/utils.ts

```diff
@@ -433,20 +431,15 @@ export function isPromiseLike<T = any>(
  * @internal
  *
  * @param command - the command on which to apply collation
- * @param target - target of command
+ * @param topology - topology of command
  * @param options - options containing collation settings
  */
 export function decorateWithCollation(
   command: Document,
-  target: { s: { topology: Topology } } | { topology: Topology },
+  topology: Topology | undefined,
```

Comment:
I think we're actually looking for something like `target: MongoClient | Db | Collection`. You should be able to reuse `getTopology` here